mono/mini/mini-profiler.c

```diff
@@ -49,39 +50,49 @@ emit_fill_call_ctx (MonoCompile *cfg, MonoInst *method, MonoInst *ret)
 void
 mini_profiler_emit_enter (MonoCompile *cfg)
 {
-	if (!MONO_CFG_PROFILE (cfg, ENTER) || cfg->current_method != cfg->method)
+	gboolean trace = mono_jit_trace_calls != NULL && mono_trace_eval (cfg->method);
+
+	if ((!MONO_CFG_PROFILE (cfg, ENTER) || cfg->current_method != cfg->method) && !trace)
 		return;
 
 	MonoInst *iargs [2];
 
 	EMIT_NEW_METHODCONST (cfg, iargs [0], cfg->method);
 
-	if (MONO_CFG_PROFILE (cfg, ENTER_CONTEXT) && !cfg->llvm_only)
+	if ((MONO_CFG_PROFILE (cfg, ENTER_CONTEXT) && !cfg->llvm_only) || trace)
```

Comment:
Shouldn't we check `cfg->llvm_only` also in the trace case? Otherwise for the LLVM-only case, we will emit an opcode the LLVM backend doesn't understand.